src/sentry/static/sentry/app/components/todos.jsx

```diff
@@ -0,0 +1,318 @@
+import moment from 'moment';
+import React from 'react';
+import {t, tct} from '../locale';
+
+import ApiMixin from '../mixins/apiMixin';
+import OrganizationState from '../mixins/organizationState';
+
+const TodoItem = React.createClass({
+  propTypes: {
+    task: React.PropTypes.object,
+  },
+
+  mixins: [OrganizationState],
+
+  getInitialState: function() {
+    return {
+      showConfirmation: false
+    };
+  },
+
+  toggleConfirmation: function() {
+    this.setState({showConfirmation: !this.state.showConfirmation});
+  },
+
+  skip: function(task) {
+    this.props.onSkip(task);
+    this.setState({showConfirmation: false});
+  },
+
+  render: function() {
+    let org = this.getOrganization();
+
+    let classNames = '';
+    let description;
+
+    switch(this.props.task.status) {
+      case 'Complete':
+        classNames += ' checked';
+        description = tct('[user] completed [dateCompleted]', {
+          user: this.props.task.user,
+          dateCompleted: moment(this.props.task.dateCompleted).fromNow(),
+        });
+        break;
+      case 'Pending':
+        classNames += ' pending';
+        description = tct('[user] kicked off [dateCompleted]', {
+          user: this.props.task.user,
+          dateCompleted: moment(this.props.task.dateCompleted).fromNow(),
+        });
+        break;
+      case 'Skipped':
+        classNames += ' skipped';
+        description = tct('[user] skipped [dateCompleted]', {
+          user: this.props.task.user,
+          dateCompleted: moment(this.props.task.dateCompleted).fromNow(),
+        });
+        break;
+      default:
+        description = this.props.task.description;
+    }
+
+    if (this.state.showConfirmation) {
+      classNames += ' blur';
+    }
+
+    let learn_more_url = '';
+    if (this.props.task.feature_location === 'project') {
+      learn_more_url = '/organizations/' + org.slug + '/projects/choose/?next=' + this.props.task.location;
+    } else if (this.props.task.feature_location === 'organization') {
+      learn_more_url = '/organizations/' + org.slug + '/' + this.props.task.location;
+    } else if (this.props.task.feature_location === 'absolute') {
+      learn_more_url = this.props.task.location;
+    }
+
+    let showSkipButton = this.props.task.skippable && this.props.task.status != 'Skipped' &&
+      this.props.task.status != 'Complete' && !this.state.showConfirmation;
+
+    return (
+      <li className={classNames}>
+        { this.props.task.status == 'Pending' && <span className="pending-bar" /> }
+        <div className="todo-content">
+          <div className="ob-checkbox">
+            { this.props.task.status == 'Complete' && <span className="icon-checkmark" /> }
```

Comment:
i would quickly note we prefer lowercase names for things like this, unless we specifically call it a label (but that'd be another thing). Probably not a massive change, and we could always change it later (though it will get trickier after release)